ps_mbo.php

```diff
@@ -16,898 +16,382 @@
+        // has to be loaded in header to prevent flash of content
+        $this->context->controller->addJs($this->getPathUri() . 'views/js/recommended-modules.js?v=' . $this->version);
+
+        if ($this->shouldAttachRecommendedModulesButton()
+            || $this->shouldAttachRecommendedModulesAfterContent()
+        ) {
+            $this->context->controller->addCSS($this->getPathUri() . 'views/css/recommended-modules.css');
+            $this->context->controller->addJs(
+                str_ireplace(
+                    _PS_CORE_DIR_,
+                    '',
+                    _PS_BO_ALL_THEMES_DIR_
                 )
+                . 'default/js/bundle/module/module_card.js?v='
+                . _PS_VERSION_
             );
         }
-
-        return array(
-            'connected' => false,
-            'login_url' => $container->get('router')->generate(
-                'admin_addons_login',
-                [],
-                UrlGeneratorInterface::ABSOLUTE_URL
-            )
-        );
     }
 
     /**
-     * uninstall()
+     * Hook displayDashboardTop.
+     * Includes content just below the toolbar.
      *
-     * @param none
-     * @return bool
+     * @return string
      */
-    public function uninstall()
+    public function hookDisplayDashboardTop()
     {
-        // unregister hook
-        if (!parent::uninstall()) {
-            $this->_errors[] = $this->trans('There was an error during the uninstallation.', [], 'Admin.Modules.Notification');
-            return false;
-        }
-
-        $idTab = Tab::getIdFromClassName('AdminModulesCatalog');
-        if ($idTab !== false) {
-            $catalogTab = new Tab($idTab);
-            $catalogTab->active = true;
-            $catalogTab->save();
-        }
-
-        $idTab = Tab::getIdFromClassName('AdminThemesCatalog');
-        if ($idTab !== false) {
-            $catalogTab = new Tab($idTab);
-            $catalogTab->active = true;
-            $catalogTab->save();
-        }
-
-        $idTab = Tab::getIdFromClassName('AdminPsMboTheme');
-        if ($idTab !== false) {
-            $catalogTab = new Tab($idTab);
-            $catalogTab->delete();
-        }
-
-        $idTab = Tab::getIdFromClassName('AdminPsMboModule');
-        if ($idTab !== false) {
-            $catalogTab = new Tab($idTab);
-            $catalogTab->delete();
-        }
+        /** @var UrlGeneratorInterface $router */
+        $router = SymfonyContainer::getInstance()->get('router');
+        $this->smarty->assign([
+            'shouldAttachRecommendedModulesAfterContent' => $this->shouldAttachRecommendedModulesAfterContent(),
+            'shouldAttachRecommendedModulesButton' => $this->shouldAttachRecommendedModulesButton(),
+            'shouldUseLegacyTheme' => $this->isAdminLegacyContext(),
+            'recommendedModulesTitleTranslated' => $this->trans('Recommended Modules and Services'), // Retrieved from messages.xlf
```

Comment:
```suggestion
            'recommendedModulesTitleTranslated' => $this->trans('Recommended Modules and Services'),
```

Reply:
Done, this comment was to explain why no translation domain is used.